network.go

```diff
@@ -1128,16 +1099,29 @@ func createEndpointsFromScan(networkNSPath string) ([]Endpoint, error) {
 	}
 	defer netlinkHandle.Delete()
 
-	netInfoList, err := networkInfoListFromNetworkScan(netlinkHandle)
+	linkList, err := netlinkHandle.LinkList()
 	if err != nil {
 		return []Endpoint{}, err
 	}
 
 	idx := 0
-	for _, netInfo := range netInfoList {
+	for _, link := range linkList {
```

Comment:
This looks good, thanks for the optimization here!
We might need to break `createEndpointsFromScan()` into subfunctions because it seems to me that we're close to the `gocyclo 15` limit.

Reply:
Looks like its at 11 with this change.  There'll be more changes later on which should simplify it (I think one function for determining endpoint type - whether it be physical, virtual or vhost-user based), a case instead of if/else..

Reply:
Ok !